pkg/files/directory.go

```diff
@@ -0,0 +1,77 @@
+package files
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+type DirectoryMap struct {
+	Root   string  `json:"root"`
+	Routes []Route `json:"routes"`
+}
+
+type Route struct {
+	Path  string   `json:"path"`
+	Files []string `json:"files"`
+}
+
+// Takes a path to a directory and returns a list of folders and files in there.
+// The folders are absolute paths based on the root provided.
+func WalkDirectory(root string) (folders []string, files []string, err error) {
+	dir, err := os.ReadDir(root)
+	if err != nil {
+		return folders, files, fmt.Errorf("failed to read directory %s: "+err.Error(), root)
+	}
+
+	for _, elem := range dir {
+		name := elem.Name()
+		if elem.IsDir() {
+			folders = append(folders, root+name)
+		} else {
+			files = append(files, name)
+		}
+	}
+
+	return folders, files, nil
+}
```

Comment:
You might be able to use https://pkg.go.dev/path/filepath#Walk for recursive walking.

Reply:
It seems the walk function uses the path string from the os. Which in testing causes it to give path strings like `"..\\..\\_local_dev\\login"` on windows. Will need some further investigation on the best approach as this will probably need to be in the same format, no matter the underlying os.